action/dashboard/update.go

```diff
@@ -0,0 +1,125 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package dashboard
+
+import (
+	"fmt"
+	"slices"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/go-vela/sdk-go/vela"
+	api "github.com/go-vela/server/api/types"
+	"github.com/go-vela/server/constants"
+)
+
+// Update modifies a dashboard based off the provided configuration.
+func (c *Config) Update(client *vela.Client) error {
+	logrus.Debug("executing update for dashboard configuration")
+
+	dashCard, _, err := client.Dashboard.Get(c.ID)
+	if err != nil {
+		return err
+	}
+
+	// pull dashboard metadata from the API response
+	dashboard := dashCard.Dashboard
+
+	// drop specified repositories from the dashboard
+	if len(c.DropRepos) > 0 {
+		newRepos := []*api.DashboardRepo{}
+
+		for _, r := range dashboard.GetRepos() {
+			if !slices.Contains(c.DropRepos, r.GetName()) {
+				newRepos = append(newRepos, r)
+			}
+		}
+
+		dashboard.SetRepos(newRepos)
+	}
+
+	// add specified repositories from the dashboard
+	if len(c.AddRepos) > 0 {
+		repos := dashboard.GetRepos()
+
+		for _, r := range c.AddRepos {
+			repo := new(api.DashboardRepo)
+			repo.SetName(r)
+
+			if len(c.Branches) > 0 {
+				repo.SetBranches(c.Branches)
+			}
+
+			if len(c.Events) > 0 {
+				repo.SetEvents(c.Events)
+			}
+
+			repos = append(repos, repo)
+		}
+
+		dashboard.SetRepos(repos)
+	}
+
+	// update specified repositories from the dashboard
+	if len(c.TargetRepos) > 0 {
+		repos := dashboard.GetRepos()
+		for _, r := range repos {
+			if slices.Contains(c.TargetRepos, r.GetName()) {
+				if len(c.Branches) > 0 {
+					r.SetBranches(c.Branches)
+				}
+
+				if len(c.Events) > 0 {
+					r.SetEvents(c.Events)
+				}
+			}
+		}
+
+		dashboard.SetRepos(repos)
+	}
+
+	// drop specified admins from the dashboard
+	if len(c.DropAdmins) > 0 {
+		newAdmins := []*api.User{}
+
+		for _, a := range dashboard.GetAdmins() {
+			if !slices.Contains(c.DropAdmins, a.GetName()) {
+				newAdmins = append(newAdmins, a)
+			}
+		}
```

Comment:
do we need a check to make sure that there is at least one admin left and/or add server side?

Reply:
https://github.com/go-vela/server/blob/8a744e48d224197478bb48a468186783b04c44db/api/dashboard/create.go#L144-L167 This function uses the caller as a starting point for any `admins` input.

Reply:
aha! i looked at update and didn't drill into that function. so, caller is always in it. someone amazing must have added that 😺